app/router.js

```diff
@@ -20,4 +22,42 @@ router.get('/:id', (req, res) => {
   return res.send(layout({ title: `Widget #${widget.id}`, widget }, show));
 });
 
+function validateForm(req, res, widget) {
+  let name = req.body.name;
+  let purpose = req.body.purpose;
+  let active = req.body.active;
+
+  // Validate name is not duplicate
+  let isDuplicate = widget.validateForDuplicateName()
+  if (isDuplicate) {
+    return res.send(
```

Comment:
this flow will only return the first error it encounters.  Can you have all errors returned to the front-end?

Reply:
Created an array to accumulate errors as they build up.

All errors should return on the frontend as well with a new line for each error.